src/components/common/ClusterOverlapsTable.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { Component } from 'react';
```

Comment:
After writing this up I now think you should wait to make this change until either Max or I merge ours because it depends on some things we've already done. What do you think @dweemx ?

Could you please refactor this as follows (rough outline, details are up to you):
1. Create a directory `src/components/ClusterOverlaps`

2. In `src/components/ClusterOverlaps/constants.ts`:
```typescript
import ColumnHeaderFragment from './ColumnHeaderFragment';

export const NAME = 'clusterOverlaps';

export const columnsText: Record<ColumnIdentifier, string> = {
        clustering_name: 'Clustering',
        cluster_name: 'Cluster',
        n_cells: '# Cells',
        cells_in_cluster: '% Cells',
        cluster_in_cells: '% Cluster',
    };

export const columnsHeaders = {
        ...columnsText,
        cells_in_cluster: ColumnHeaderFragment('% Cells', 'The percentage of the selected cells within the cluster'),
        cluster_in_cells: ColumnHeaderFragment('% Cluster', 'The percentage of the cluster within the selection')
};
```

3. In `src/components/ClusterOverlaps/types.ts`:
```typescript
import { columnsText, columnsHeaders } from './constants';

export interface SortOptions {
    id: string;
    desc: boolean;
}

export interface ClusterOverlaps {
    clustering_name: string;
    cluster_name: string;
    n_cells: number;
    cells_in_cluster: number;
    cluster_in_cells: number;
}

export interface Column {
    Header: string;
    accessor: string;
    style: { whiteSpace: string };
    sortMethod: (x: number, y: number) => number;
    width: number;
}

export toColumn = (name: string, clusterOverlaps: ClusterOverlaps): Column => {
        const sort = [
            'n_cells',
            'cells_in_cluster',
            'cluster_in_cells',
        ].includes(name);

        return {
            Header: columnsHeaders[name],
            accessor: name,
            style: { whiteSpace: 'unset' },
            sortMethod: sort
                ? R.comparator((a, b) => number(a) > number(b))
                : undefined,
            width: sort
                ? this.getColumnWidth(
                      clusterOverlaps,
                      name,
                      columnsText[name]
                  )
                : undefined,
        };
};
```

4. In `src/components/ClusterOverlaps/ColumnHeaderFragment.tsx`:
```typescript
import React from 'react';

export default const ColumnHeaderFragment = (name: string, popupContent: string) => {
            <React.Fragment>
                {name}
                <br />
                <Popup
                    basic
                    content={popupContent}
                    position='top left'
                    trigger={
                        <Icon
                            name='question circle'
                            style={{ display: 'inline' }}
                            className='pointer'
                        />
                    }
                />
            </React.Fragment>
};
```

5. In `src/components/ClusterOverlaps/ClusterOverlapsTable.tsx`:
```typescript
import React from 'react';
import ReactTable from 'react-table';
import * as R from 'ramda';
import { Icon, Popup } from 'semantic-ui-react';
import { connect } from 'react-redux';

import { toColumn, SortOptions } from './types';
import { columnsText } from './constants';
import * as Action from './actions';

interface ClusterOverlapsTableProps {
    clusterOverlaps: ClusterOverlaps[];
    sortOptions: Array<SortOptions>;
    sortChange: (val) => Action.SortChange;
}

const ClusterOverlapsTable: React.FC<ClusterOverlapsTableProps> = (props) => {
    const clusterOverlapsColumns = R.map(this.toColumn, R.keys(this.columnsText));

    const onSortChange = (val) => {

    };

    return (
        <ReactTable
            data={props.clusterOverlaps}
            sorted={props.sortOptions}
            onSortedChange={props.sortChange}
            columns={[
                {
                    Header: 'Selection/Cluster Overlaps',
                    columns: clusterOverlapsColumns,
                },
            ]}
            pageSizeOptions={[3, 5, 10, 15]}
            defaultPageSize={3}
            className='-striped -highlight'
        />
    );
};

const mapStateToProps = (state: RootState) => {
    return {
        sortOptions: state[NAME].sortOptions;
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        sortChange: (val) => dispatch(Action.sortChange(val)),
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(ClusterOverlapsTable);
```

6. In `src/components/ClusterOverlaps/actionTypes.ts`:
```typescript
export const SORT_CHANGE = 'clusterOverlaps/SORT_CHANGE';
```

7. In `src/components/ClusterOverlaps/actions.ts`:
```typescript
import * as ActionType from './actionTypes';

export interface SortChange {
    type: typeof ActionType.SORT_CHANGE,
    payload: unknown // whatever val is on line 147, 148
}

export const sortChange = (val): SortChange => {
    return {
        type: ActionType.SORT_CHANGE,
        payload: val,
    };
};

export type ClusterOverlapAction = SortChange;
```

8. In `src/components/ClusterOverlaps/model.ts`:
```typescript
import { SortOptions } from './types';

export type State {
    sortOptions: Array<SortOptions>;
};
```

9. In `src/components/ClusterOverlaps/reducer.ts`:
```typescript
import produce, { Draft } from 'immer';

import { ClusterOverlapAction } from './actions';
import * as ActionType from './actionTypes';
import { State } from './model';

const initialState: State = [
    { id: 'cells_in_cluster', desc: false },
    { id: 'cluster_in_cells', desc: false },
];

const reducer = produce((draft: Draft<State>, action: ClusterOverlapAction) => {
    switch (action.type) {
        case ActionType.SORT_CHANGE:
            draft.sortOptions = action.payload;
            break;
    }
}, initialState);

export { reducer };
```

10. In `src/components/ClusterOverlaps/index.ts`:
```typescript

import * as Actions from './actions';
import ClusterOverlapsTable from './ClusterOverlapsTable';
import * as Constants from './constants';
import { reducer } from './reducer';

export default {
    Actions,
    ClusterOverlapsTable,
    Constants,
    reducer,
};
```

11. And finally, in ` src/redux/reducers/index.ts `:
Add:
```typescript
import { reducer as ClusterOverlapsReducer } from '../../components/ClusterOverlaps/reducer';
import * as ClusterOverlaps from '../../components/ClusterOverlaps/constants';

//...
const rootReducer = combineReducers({
    ['main']: main,
    [ClusterOverlaps.NAME]: ClusterOverlapsReducer,
});
```

Reply:
@MaybeJustJames indeed it makes sense to wait that the 2 current PR with Redux stuff have been merged

Reply:
As per discussions, this will be postponed.